src/components/implementation/tests/unit_pmu/unit_pmu.c

```diff
@@ -0,0 +1,26 @@
+#include <cos_kernel_api.h>
+#include <cos_types.h>
+#include <hw_perf.h>
+
+#define NUM_PAGES 12
+
+word_t buf[NUM_PAGES*PAGE_SIZE];
+
+int
+main()
+{
+	printc("Running pmu test...\n");
+
+	unsigned long dtlb_misses = hw_perf_cnt_dtlb_misses();
+	
+	for (int i = 0; i < NUM_PAGES; i++) {
+		buf[i * PAGE_SIZE] = 0x01;
+	}
+
+	dtlb_misses = hw_perf_cnt_dtlb_misses() - dtlb_misses;
+	printc("DTLB MISSES %lu\n", dtlb_misses);
```

Comment:
Can you do a SUCCESS/FAILURE, but just making sure it detects a number of TLBs within 50% of the right number? Just a rough test is fine. Better than nothing. Can still print out the misses as you are here.